### DISCUSSION TITLE
Feature Requests.

### DISCUSSION THREAD

**Discussion by SuhasDissa** [you-apps/WallYou#133] [Ideas]
Hi everyone,

We're starting this thread to help keep the issue log clean and focused on the most critical issues. Please use this thread to post your feature requests and design suggestions.

We appreciate your feedback!

**Comment by daephx** [comment]
Just found this app, I've considered learning android development just to make into my own ideal app for handling wallpapers and this covers a number of things I've desired, open source, auto-changer, access to online providers, basic manager/favorites!

Good work with this thus far!

Here are some features which I would like to see as well.
Take what you will, the latter few might be nitpicks but thanks in advance for listening and considering these improvements.

- Better support for local wallpapers, at lease add a local directory to the sidebar.
- As a new user, the current method for selecting changer categories is not immediately clear.
  It took me longer than I'd like to admit to find the currently provided local directory option.
- With the above two. Allow local favorites, such that the changer can use online and local collections simultaneously.
- Allow the changer to be disabled/locked/manually overwritten for the home/lock screen separately,
  so that the changer works while at least one of the two screen remains static.
- Images thumbnails are blurry in certain providers.
  For the thumbnails I think you can get away with having an option for more rows/columns to make this less perceptible.
- Scrolling through large thumbnails feels a bit weird. Part of it is always cut off, it takes multiple flicks to get to the favorite button.
  If not more rows/columns maybe use snap scrolling to each image?

<details>

<summary>Due to length, I've detailed much of my reasoning here:</summary>
</br>

I feel the local options are lacking. For being "Privacy focused" it leans too much into online providers and doesn't allow for suitable management of local wallpapers. I did see a relevant issue regarding this, the gallery app exists, but it's missing some features that this app could provide, namely local favorites for the changer. I just personally think it's a poor choice to almost entirely forgo local browsing.

There should at least be an option in the sidebar for local files. Perhaps make it default to `/Pictures/Wallpapers/` but this way you could browse wallpapers from providers and maybe you just want to save it but not make it your background? Cool, throw it into the local collection. Not sure how you currently handle caching but that would help to make the app work offline.

With that said, it also took me way too long to figure out that you can even select a local folder via the changer options. For a minute I thought this app was 100% online providers and I was about to uninstall until I found someone mention it in an issue. Where the option is located with the side scrolling list of providers, it wasn't very clear that you have to click above them to get the other categories `Online`, `Favorites`, `Local`.
Of these categories, It would be nice if they weren't mutually exclusive, just let the the randomizer pick one from local or query a provider, if not that than let favorites be where they both merge.

I'm also surprised that there is no way to disable the home screen or lock screen individually or otherwise specify that one should be handled manually. You can combine them, set them to local or online separately, solid additions! But I tend to use the same lockscreen (a plain black image), maybe you want artwork on the lockscreen to be automatically shuffled but the home screen is always your pet/s, kid/s, spouse, waifu. Whatever your preference! However, that doesn't appear to be an option at the moment.

</details>

**Reply by Bnyro** [reply]
The thing with not being able to disable the wallpaper changer for either the lock screen or the home screen is a good point, I definitely agree that we should add this.

Local wallpapers for the wallpaper changer are only a side affect and never really the goal of the app, I'm not really certain if adding a dedicated tab and favorites for that would actually be a good idea given the fact that you should probably use a gallery app or file manager for managing local photos.

**Reply by daephx** [reply]
I would honestly use the gallery/file browser for moving/deleting or otherwise properly managing files.
But I like the idea of having favorites, collections, search by color pallette closer to the auto-changer.
As my phone (android 7) provides some of these features but is missing the big ones, the auto-changer and its ease-of-access to online providers.

Regarding scope, if you need to put your foot down somewhere and that's "here", then I understand.
My personal philosophy is just more inline with local being a first class citizen, but it's not my project.

**Reply by kurtph1969** [reply]
Hi, just discovered this excellent app. Question: would it be possible to display local files as a source please? I would like the possibility to add local images to favorites but I couldn't see a way to do this. Thank you.

**Comment by Hipopotopulus** [comment]
Can you please make a shorter interval for the wallpaper to change. Like every minute or every time the phone is unlocked. For now it's 1 hour minimum. Thx in advance

**Reply by Bnyro** [reply]
see https://github.com/you-apps/WallYou/issues/229

**Reply by Hipopotopulus** [reply]
Oh, thank you

**Comment by MEntOMANdo** [comment]
SmartLauncher (www.smartlauncher.net) provides some nice visual effects behind the launcher widgets, folders, etc. But in order to apply those effects, it needs a copy of the current wallpaper image. 

They have implemented an API to notify SmartLauncher that a new wallpaper has been applied. Wall You has far more robust features when it comes to wallpaper, but it would be nice if Wall You could implement their "Smart Launcher Wallpaper API" to notify when the home page (not lock screen) wallpaper changes. 

Details about this API [are located here](https://docs.smartlauncher.net/work-with-smart-launcher/wallpaper-apis); it's a single call just before applying the wallpaper. Of course, ideally this would need to be an option so that it only affects users who have SmartLauncher installed.

**Reply by Bnyro** [reply]
Unfortunately Smart Launcher doesn't seem to be open source, so I think that this is not somethingthat will benefit most users as most users probably use open source launchers (e.g. Kvaesito).

I'll re-consider this if enough people are interested in it, but for now this seems rather like a niche feature.

**Comment by anarkia47** [comment]
Could you add a widget that allows users to instantly change the wallpaper with a single tap, similar to the existing quick tile shortcut? This would make switching wallpapers even faster and more convenient.
